skimage/util/dtype.py

```diff
@@ -465,10 +467,36 @@ def img_as_float(image, force_copy=False):
     and can be outside the ranges [0.0, 1.0] or [-1.0, 1.0].
 
     """
-    return _convert(image, np.floating, force_copy)
+    return _img_as_float(image, force_copy, preserve_range=False)
```

Comment:
I don't understand this change. Why did you wrap `_convert` with the new function `_img_as_float`?

Also one could argue that it would be more explicit to use `dtype=np.floating` here and not rely on `_img_as_float`'s implicit behavior for `dtype=None`. Or rather use `np.floating` as the default in the function signature?

Reply:
not sure why I did that, will take a look again.

Reply:
`_img_as_float` has been removed and `_convert` is just used directy now.